tests/utils/test_yaml.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+
+from kpops.utils.yaml import substitute_nested
+
+
+@pytest.mark.parametrize(
+    ("input", "substitution", "expected"),
+    [
+        pytest.param(
+            '"a": "b", "${a}": "c", "a_${b}": "d", "e": "${a_${b}}"',
+            {"a": "b", "${a}": "c", "a_${b}": "d", "e": "${a_${b}}"},
+            '"a": "b", "b": "c", "a_c": "d", "e": "d"',
+            id="requires-multiple-passes",
```

Comment:
nice 🤩  I didn't know about these. does it print the id if a certain test case fails?

Reply:
It does :)